llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -226,23 +220,22 @@ Expected<std::string> compileGlsl(const std::string &inFilename, ShaderStage *st
 
   const unsigned *spvBin = nullptr;
   unsigned binSize = spvGetSpirvBinaryFromProgram(program, 0, &spvBin);
-  fwrite(spvBin, 1, binSize, outFile);
 
   textSize = binSize * 10 + 1024;
   std::vector<char> spvText(textSize, '\0');
-  LLPC_OUTS("\nSPIR-V disassembly: " << outFilename << "\n");
+  LLPC_OUTS("\nSPIR-V disassembly:\n");
   spvDisassembleSpirv(binSize, spvBin, textSize, spvText.data());
   LLPC_OUTS(spvText.data() << "\n");
 
-  return outFilename;
+  return BinaryData{static_cast<size_t>(binSize), spvBin};
```

Comment:
Can you explain who is responsible for freeing the memory behind spvBin? Would it be possible to return a vector instead?

Reply:
Added a comment